lib/active_merchant/billing/gateways/pay_hub.rb

```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
+        
+        schedule[:specific_dates_schedule] = specific_dates_schedule
       end
 
-      def add_customer_data(post, options = {})
-        post[:first_name] = options[:first_name]
-        post[:last_name] = options[:last_name]
-        post[:phone] = options[:phone]
-        post[:email] = options[:email]
+      def weekly_schedule(schedule, options = {})
+        weekly_schedule = {}
+        weekly_schedule[:weekly_bill_days] = options[:weekly_bill_days]
+        
+        schedule[:weekly_schedule] = weekly_schedule
       end
 
-      def add_address(post, address)
-        return unless address
-        post[:address1] = address[:address1]
-        post[:address2] = address[:address2]
-        post[:zip] = address[:zip]
-        post[:state] = address[:state]
-        post[:city] = address[:city]
+      def monthly_schedule(schedule, options = {})
+        monthly_schedule = {}
+        case options[:monthly_type]
+          when "O"
+            monthly_schedule[:monthly_each_days] = options[:monthly_each_days]
+          when "E"
+            monthly_schedule[:monthly_on_the_day_of_week_in_month ] = options[:monthly_on_the_day_of_week_in_month ]
+        end
+          
+        schedule[:monthly_schedule] = monthly_schedule  
       end
 
-      def add_amount(post, amount)
-        post[:amount] =  amount(amount)
+      def yearly_schedule(schedule, options = {})
+        yearly_schedule  = {}
+        yearly_schedule[:year_to_start] = options[:year_to_start]
+        yearly_schedule[:yearly_bill_on_day_of_month ] = options[:yearly_bill_on_day_of_month]
+        
+        schedule[:yearly_schedule] = yearly_schedule  
       end
 
-      def add_creditcard(post, creditcard)
-        post[:cc] = creditcard.number
-        post[:month] = creditcard.month.to_s
-        post[:year] = creditcard.year.to_s
-        post[:cvv] = creditcard.verification_value
+      def schedule_start_and_end(schedule, options = {})
+        schedule_start_and_end  = {}
+        schedule_start_and_end[:start_date] = options[:start_date]
+        case options[:end_date_type]
+          when "A"
+            schedule_start_and_end[:end_after_bill_count] =  options[:end_after_bill_count]
+          when "O"
+            schedule_start_and_end[:end_date] = options[:end_date]
+        end  
+        schedule[:schedule_start_and_end] = schedule_start_and_end
       end
 
       def parse(body)
         JSON.parse(body)
       end
 
-      def commit(post)
-        success = false
+      def headers(options = {})
+        oauth_token = options[:oauth_token] || @options[:oauth_token]
+
+        headers = {
+          "Authorization" => "Bearer " + oauth_token,
+          "Content-Type" => "application/json",
+          "Accept" => "application/json"
+        }
+        headers
+      end
+
+      def api_version(options)
+        options[:version] || @options[:version] || "v2"
+      end
+      
+      def commit(method, endpoint, post, options)
```

Comment:
You've added this `method` param, but never actually use it. Everything uses `ssl_post`